StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CardBrandFilter.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  CardBrandFilter.swift
+//  StripePaymentSheet
+//
+//  Created by Nick Porter on 10/10/24.
+//
+
+import Foundation
+@_spi(STP) import StripeCore
+
+struct CardBrandFilter {
+
+    static let `default`: CardBrandFilter = .init(cardBrandAcceptance: .all)
+
+    private let cardBrandAcceptance: PaymentSheet.CardBrandAcceptance
+
+    init(cardBrandAcceptance: PaymentSheet.CardBrandAcceptance) {
+        self.cardBrandAcceptance = cardBrandAcceptance
+    }
+
+    /// Determines if a merchant can accept a card brand based on `cardBrandAcceptance`
+    /// - Parameter cardBrand: The `STPCardBrand` to determine if acceptance
+    /// - Returns: Returns true if this merchant can accept this card brand, false otherwise
+    func isAccepted(cardBrand: STPCardBrand) -> Bool {
+        switch cardBrandAcceptance {
+        case .all:
+            return true
+        case .allowed(let allowedCardBrands):
+            // If a merchant has specified a list of brands to allow, block unknown brands
+            guard let brandCategory = cardBrand.asBrandCategory else {
```

Comment:
This looks like it will log many times: Continuously as the user types a card number, and for each enumeration in the `STPCardBrand.allCases.filter { cardFilter.isAccepted(cardBrand: $0) }` code below. Maybe we could have it log less often, maybe just from the PaymentSheet UI somehow when the card number entry is completed?

Reply:
You're right! That's gonna be super spammy! I changed this to only log from the PAN element, and not log if it's the same disallowed brand.